test/legacy/21-supervisor-api.spec.ts

```diff
@@ -1,288 +0,0 @@
-			// multi-app scoped key...
-			const multiAppScopedKey = await deviceApi.generateScopedKey(1, 'other', {
-				scopes: [1, 2].map((appId) => {
-					return { type: 'app', appId };
-				}),
-			});
-
-			await request
-				.get('/v2/applications/1/state')
-				.set('Accept', 'application/json')
-				.set('Authorization', `Bearer ${multiAppScopedKey}`)
-				.expect(200);
-
-			await request
-				.get('/v2/applications/2/state')
-				.set('Accept', 'application/json')
-				.set('Authorization', `Bearer ${multiAppScopedKey}`)
-				.expect(200);
-		});
-		it('should generate a key which is scoped for all applications', async () => {
-			await request
-				.get('/v2/applications/1/state')
-				.set('Accept', 'application/json')
-				.set('Authorization', `Bearer ${await deviceApi.getGlobalApiKey()}`)
-				.expect(200);
-
-			await request
-				.get('/v2/applications/2/state')
-				.set('Accept', 'application/json')
-				.set('Authorization', `Bearer ${await deviceApi.getGlobalApiKey()}`)
-				.expect(200);
-		});
-		it('should have a cached lookup of the key scopes to save DB loading', async () => {
-			const scopes = await deviceApi.getScopesForKey(
-				await deviceApi.getGlobalApiKey(),
-			);
-
-			const key = 'not-a-normal-key';
-			await db.initialized();
-			await db
-				.models('apiSecret')
-				.update({
-					key,
-				})
-				.where({
-					key: await deviceApi.getGlobalApiKey(),
-				});
-
-			// the key we had is now gone, but the cache should return values
-			const cachedScopes = await deviceApi.getScopesForKey(
-				await deviceApi.getGlobalApiKey(),
-			);
-			expect(cachedScopes).to.deep.equal(scopes);
-
-			// this should bust the cache...
-			await deviceApi.refreshKey(await deviceApi.getGlobalApiKey());
-
-			// the key we changed should be gone now, and the new key should have the cloud scopes
-			const missingScopes = await deviceApi.getScopesForKey(key);
-			const freshScopes = await deviceApi.getScopesForKey(
-				await deviceApi.getGlobalApiKey(),
-			);
-
-			expect(missingScopes).to.be.null;
-			expect(freshScopes).to.deep.equal(scopes);
-		});
-		it('should regenerate a key and invalidate the old one', async () => {
-			// single app scoped key...
-			const appScopedKey = await deviceApi.generateScopedKey(1, 'main');
-
-			await request
-				.get('/v2/applications/1/state')
-				.set('Accept', 'application/json')
-				.set('Authorization', `Bearer ${appScopedKey}`)
-				.expect(200);
-
-			const newScopedKey = await deviceApi.refreshKey(appScopedKey);
-
-			await request
-				.get('/v2/applications/1/state')
-				.set('Accept', 'application/json')
-				.set('Authorization', `Bearer ${appScopedKey}`)
-				.expect(401);
-
-			await request
-				.get('/v2/applications/1/state')
-				.set('Accept', 'application/json')
-				.set('Authorization', `Bearer ${newScopedKey}`)
-				.expect(200);
-		});
-	});
-
-	describe('State change logging', () => {
```

Comment:
There was a suite for API startup & shutdown logging in the legacy tests, but I don't think the test suite is particularly useful so deleted it entirely instead of moving it to unit/integration.